packages/components/src/internal/components/base/Grid.tsx

```diff
@@ -379,6 +380,14 @@ export const Grid: FC<GridProps> = memo(props => {
     const gridData = processData(data);
     const gridColumns = columns !== undefined ? processColumns(columns) : resolveColumns(gridData);
 
+    const divRef = useRef<HTMLDivElement>();
+    useEffect(() => {
+        if (!fixedHeight) return;
+        const maxHeight = window.innerHeight * 0.7;
+        divRef.current.style.height =
+            divRef.current.lastElementChild?.clientHeight < maxHeight ? 'unset' : maxHeight + 'px';
+    }, [fixedHeight, gridData.size]);
```

Comment:
nit: Doesn't look like `gridData.size` needs to be included here. Is there an extraneous reason it is includeded?

Reply:
I want to trigger a recalculation of the div height each time that the number of rows in the grid changes, so I added `gridData.size` for that reason. I should add a comment to that affect.